cfme/common/vm.py

```diff
@@ -774,6 +774,18 @@ def _paused():
         else:
             raise ValueError("Invalid state '{}'".format(state))
 
+    def cleanup_on_provider(self):
+        """
+        Method to remove a VM from the provider after tests.
+
+        Checks that the VM exists on the provider, ensures it is in 'powered off' state,
+        and deletes it. Any exceptions raised during delete will be logged only.
+
+        This method may be overriden at the provider-specific level to delete and
+        then do some additional work afterwards (like deleting resources attached to this VM)
+        """
+        self.delete_from_provider()
```

Comment:
**optional** this indirection and the following changes give me the impression that `delete_from_provider` and `cleanup_on_provider` are very similar, i wonder if taking out the former would reveal additional call sites to consider

Reply:
Agreed they are similar, I think it's a good idea to have a delete for treating purposes, and a clean up for finalizer purposes. For example you may want to delete a VM or instance but have it's disk stay present for a test, whereas "cleanup" would remove all resources tied to the VM/instance, but people will need to extend the method in the provider-specific classes

Reply:
/s/treating/testing